clientd3d/d3drender_world.h

```diff
@@ -14,8 +14,118 @@
 #ifndef _D3DRENDERWORLD_H
 #define _D3DRENDERWORLD_H
 
-void D3DRenderWorld(room_type* room, Draw3DParams* params, room_contents_node* pRNode);
-void D3DGeometryBuildNew(room_type* room, d3d_render_pool_new* pPool, bool transparent_pass);
+struct WorldCacheSystemParams {
+
+    d3d_render_cache_system* worldCacheSystem;
+    d3d_render_cache_system* worldCacheSystemStatic;
+
+    d3d_render_cache_system* lMapCacheSystem;
+    d3d_render_cache_system* lMapCacheSystemStatic;
+
+    d3d_render_cache_system* wallMaskCacheSystem; // geometry
+
+    WorldCacheSystemParams(
+        d3d_render_cache_system* worldCacheSystemParam,
+        d3d_render_cache_system* worldCacheSystemStaticParam,
+        d3d_render_cache_system* lMapCacheSystemParam,
+        d3d_render_cache_system* lMapCacheSystemStaticParam,
+        d3d_render_cache_system* wallMaskCacheSystemParam
+    )
+        : worldCacheSystem(worldCacheSystemParam),
+          worldCacheSystemStatic(worldCacheSystemStaticParam),
+          lMapCacheSystem(lMapCacheSystemParam),
+          lMapCacheSystemStatic(lMapCacheSystemStaticParam),
+          wallMaskCacheSystem(wallMaskCacheSystemParam)
+    {}
+};
+
+struct WorldPoolParams {
+
+	d3d_render_pool_new* worldPool;
+	d3d_render_pool_new* worldPoolStatic;
+
+	d3d_render_pool_new* lMapPool;
+	d3d_render_pool_new* lMapPoolStatic;
+
+	d3d_render_pool_new* wallMaskPool; // geometry
+
+    WorldPoolParams(
+        d3d_render_pool_new* worldPoolParam,
+        d3d_render_pool_new* worldPoolStaticParam,
+        d3d_render_pool_new* lMapPoolParam,
+        d3d_render_pool_new* lMapPoolStaticParam,
+        d3d_render_pool_new* wallMaskPoolParam
+    )
+        : worldPool(worldPoolParam),
+            worldPoolStatic(worldPoolStaticParam),
+            lMapPool(lMapPoolParam),
+            lMapPoolStatic(lMapPoolStaticParam),
+            wallMaskPool(wallMaskPoolParam)
+    {}
+};
+
+struct WorldPropertyParams {
+    
+    LPDIRECT3DTEXTURE9 noLookThrough;
+    LPDIRECT3DTEXTURE9 lightOrange;
+
+	WorldPropertyParams(LPDIRECT3DTEXTURE9 noLookThroughParam,
+        LPDIRECT3DTEXTURE9 lightOrangeParam,
+        player_info* playerParam)
+		: noLookThrough(noLookThroughParam),
+		  lightOrange(lightOrangeParam)
+	{}
+};
+
+struct WorldRenderParams {
+
+    LPDIRECT3DVERTEXDECLARATION9 vertexDeclaration;
+    LPDIRECT3DVERTEXDECLARATION9 vertexDeclarationSecondary;
+    d3d_driver_profile driverProfile;
+
+    const WorldCacheSystemParams& cacheSystemParams;
+    const WorldPoolParams& poolParams;
+
+    D3DMATRIX view;
+    D3DMATRIX proj;
+
+    room_type* room;
+    Draw3DParams* params;
+
+    WorldRenderParams(
+        LPDIRECT3DVERTEXDECLARATION9 vertexDeclarationParam,
+        LPDIRECT3DVERTEXDECLARATION9 vertexDeclarationSecondaryParam,
+        d3d_driver_profile driverProfileParam,
```

Comment:
This will get copied twice, once at the site of the constructor call, and once into the struct variable.  Should it be a reference?